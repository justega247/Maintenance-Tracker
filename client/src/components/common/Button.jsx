import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  const {
    text, type, className, ...rest
  } = props;
  return (
    <button className={className} type={type} {...rest}>
      {text}
    </button>
  );
};

Button.defaultProps = {
  text: 'search',
  type: 'button',
  className: '',
};

Button.propTypes = {
  text: PropTypes.string,
  type: PropTypes.string,
  className: PropTypes.string,
};

export default Button;
