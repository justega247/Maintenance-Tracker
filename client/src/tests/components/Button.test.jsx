import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../components/common/Button';

let wrapper;

describe('Button', () => {
  beforeEach(() => {
    const props = {
      text: 'search',
      type: 'button',
      className: 'big',
    };
    wrapper = shallow(<Button {...props} />);
  });

  test('should correctly render Button', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
