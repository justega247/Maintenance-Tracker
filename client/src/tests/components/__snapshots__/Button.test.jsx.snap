// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should correctly render Button 1`] = `
<button
  className="big"
  type="button"
>
  search
</button>
`;
